plugins/lacrosse-scoreboard/game_renderer.py

```
@@ -619,32 +625,282 @@ def _draw_recent_game_status(self, draw: ImageDraw.Draw, game: Dict) -> None:
+            start = raw if isinstance(raw, datetime) else datetime.fromisoformat(
+                str(raw).replace("Z", "+00:00"))
+            return self._WEEKDAY_ABBR[start.astimezone(self._card_tzinfo()).weekday()]
+        except (ValueError, TypeError):
+            return ""
+
+    def _card_tzinfo(self):
+        """Timezone for weekday/24h conversions; falls back to UTC."""
+        try:
+            configured = (self.config or {}).get("timezone")
+            if configured:
+                return ZoneInfo(configured)
```

Ruff (0.16.1)

[error] Undefined name `ZoneInfo`

(F821)